lib/customisation.nix

```diff
@@ -365,6 +378,35 @@ rec {
     else
       mapAttrs mkAttrOverridable pkgs;
 
+  makeAlias =
+    let
+      currentRelease = lib.versions.majorMinor lib.version;
+    in
+    self: name:
+    {
+      type,
+      package ? null,
+      reason ? null,
+      release ? null,
+    }:
+    assert type == "alias";
+    if release == null then
+      self.${package}
+    else if release != currentRelease then
+      if package == null then
+        abort "Throw for '${name}' was added in ${release}. The alias must now be removed."
```

Comment:
A warning seems like it may be better? Then you can scoop them all up at once, and perhaps do something with `unsafeGetAttrPos` to list all the files that need changing.